optimum/intel/openvino/modeling_open_clip.py

```diff
@@ -0,0 +1,680 @@
+
+        logits_per_image = self.logit_scale.exp() * image_features @ text_features.T
+        if self.logit_bias is not None:
+            logits_per_image += self.logit_bias
+        logits_per_text = logits_per_image.T
+
+        return CLIPOutput(
+            loss=None,
+            logits_per_image=logits_per_image,
+            logits_per_text=logits_per_text,
+            text_embeds=text_features,
+            image_embeds=image_features,
+            text_model_output=text_model_outputs,
+            vision_model_output=visual_model_outputs,
+        )
+
+    def save_pretrained(
+        self,
+        save_directory: Union[str, os.PathLike],
+        push_to_hub: bool = False,
+        **kwargs,
+    ):
+        self.text_model.save_pretrained(save_directory=save_directory, push_to_hub=push_to_hub, **kwargs)
+        self.visual_model.save_pretrained(save_directory=save_directory, push_to_hub=push_to_hub, **kwargs)
+        self.config.save_pretrained(save_directory)
+        return
+
+    def push_to_hub(
+        self,
+        save_directory: str,
+        repository_id: str,
+        private: Optional[bool] = None,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+    ) -> str:
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed soon. Please use the `token` argument instead.",
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError("You cannot use both `use_auth_token` and `token` arguments at the same time.")
+            token = use_auth_token
+
+        self.text_model.push_to_hub(
+            save_directory=save_directory, repository_id=repository_id, private=private, token=token
+        )
+        self.visual_model.push_to_hub(
+            save_directory=save_directory, repository_id=repository_id, private=private, token=token
+        )
+        return
+
+    def compile(self):
+        self.text_model.compile()
+        self.visual_model.compile()
+
+    def reshape(self, batch_size: int, sequence_length: int, height: int = None, width: int = None):
+        """
+        Propagates the given input shapes on the model's layers, fixing the inputs shapes of the model.
+
+        Arguments:
+            batch_size (`int`):
+                The batch size.
+            sequence_length (`int`):
+                The sequence length or number of channels.
+            height (`int`, *optional*):
+                The image height.
+            width (`int`, *optional*):
+                The image width.
+        """
+        self.text_model.reshape(batch_size=batch_size, sequence_length=sequence_length, height=height, width=width)
+        self.visual_model.compile(batch_size=batch_size, sequence_length=sequence_length, height=height, width=width)
```

Comment:
```suggestion
        self.visual_model.reshape(batch_size=batch_size, sequence_length=sequence_length, height=height, width=width)
```

Reply:
could you add a test to verify this works as expected ?

Reply:
also we should add test for .to() and .compile()

Reply:
added